docs/sources/api/_index.md

```diff
@@ -47,6 +47,7 @@ These endpoints are exposed by the ingester:
 
 - [`POST /flush`](#post-flush)
 - [`POST /ingester/flush_shutdown`](#post-ingesterflush_shutdown)
+- [`POST /ingester/shutdown_and_forget`](#post-ingestershutdown_and_forget)
```

Comment:
`shutdown_and_forget` doesn't make it clear what it would forget. What do you think about calling it `shutdown_and_forget_ring_tokens` or `shutdown_and_remove_ring_tokens`?
We can also add a query param to `flush_shutdown` like `/ingester/flush_shutdown?forget_ring_tokens=true`.

Reply:
On the one hand I like the idea of a single endpoint, but on the other hand, I find it weird to have query parameters in URLs that only allow `POST` requests.

Reply:
It would make sense then to change the URL to `/ingester/shutdown` and make both `flush` and `delete_tokens` URL parameters.

```
/ingester/shutdown?flush=true
/ingester/shutdown?delete_tokens=true
```

(Of course keep the existing URL and deprecate it.)

What do you think @sandeepsukhani ?

Reply:
What about `/ingester/shutdown/?unregister=true`?

Keep in mind as well that documentation can clarify the purpose and mechanism for an API endpoint, if need be.

Reply:
> I find it weird to have query parameters in URLs that only allow POST requests
I agree it looks ugly but it helps us keep it tidy. If we keep adding different paths for various use cases, it would start getting messier.

I think `delete_ring_tokens` or `reset_ring_tokens` or something similar would be a bit self-descriptive.